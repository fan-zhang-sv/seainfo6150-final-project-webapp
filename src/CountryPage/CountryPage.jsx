import React, { useEffect, useState } from "react";
import PropTypes from "prop-types";
import styles from "./CountryPage.module.css";
import cx from "classnames";
import NewsCardUnderTopic from "./NewsCardUnderTopic/NewsCardUnderTopic";

import {
  ccToCountryName,
  ccToMkt,
  ccToCategory,
} from "../Modules/countryCodes";

const baseUrl =
  "https://my-news-app.cognitiveservices.azure.com/bing/v7.0/news?";

const CountryPage = (props) => {
  const cc = props.countryCode;
  const countryName = ccToCountryName[cc];
  const mkt = ccToMkt[cc];
  const categories = ccToCategory[cc];
  let [selectedCategory, setSelectedCategory] = useState(0);
  let [fetchedData, setFetchedData] = useState([]);

  useEffect(() => {
    if (props.countryCode in ccToCountryName) {
      const params = {
        mkt: mkt,
        category: categories[selectedCategory],
        originalImg: true,
      };
      const urlParams = new URLSearchParams(Object.entries(params));

      console.log("fetching data from api");
      fetch(baseUrl + urlParams, {
        headers: {
          "content-type": "application/json",
          "Ocp-Apim-Subscription-Key": "9e6eb09fc72b40d1abf23d1e5ca46a11",
        },
      })
        .then((response) => response.json())
        .then((data) => setFetchedData(data["value"]));
    }
  }, [categories, mkt, props.countryCode, selectedCategory]);

  return (
    <div className={styles.Container}>
      <div className={styles.TitleContainer}>
        <h1>{countryName + " - " + categories[selectedCategory]}</h1>
      </div>
      <div className={styles.ContentContainer}>
        <div className={styles.CategoryContainer}>
          {categories.map((category, idx) => {
            return (
              <button
                className={cx(
                  styles.CategoryButton,
                  selectedCategory === idx
                    ? styles.SelectedButton
                    : styles.PlainButton
                )}
                key={category}
                onClick={() => {
                  setSelectedCategory(idx);
                }}
              >
                {category}
              </button>
            );
          })}
        </div>
        <div className={styles.NewsContainer}>
          {fetchedData
            ? fetchedData.map((news) => (
                <NewsCardUnderTopic news={news} key={news["url"]} />
              ))
            : null}
        </div>
      </div>
    </div>
  );
};

CountryPage.propTypes = {
  countryCode: PropTypes.string.isRequired,
};

export default CountryPage;
