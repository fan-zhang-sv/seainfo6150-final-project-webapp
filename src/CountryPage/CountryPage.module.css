.Container {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.CategoryButton {
  width: 85%;
  padding: 14px;
  margin-bottom: 20px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 1rem;
}

.PlainButton {
  background-color: white;
  color: black;
  border: 2px solid #555555;
}

.SelectedButton {
  /* color: black;
  background-color: #e7e7e7; */
  border: 2px solid #555555;
  background-color: #555555;
  color: white;
}

.CategoryButton:hover {
  box-shadow: 0 6px 8px 0 rgba(0, 0, 0, 0.24), 0 17px 50px 0 rgba(0, 0, 0, 0.19);
}

.TitleContainer {
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: flex-start;
  margin-bottom: 20px;
}

.ContentContainer {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
}

.CategoryContainer {
  width: 100%;
  flex: 1 0 0;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
}

.NewsContainer {
  position: relative;
  top: -10px;
  width: 100%;
  flex: 2 0 0;
  display: grid;
  column-gap: 10px;
  row-gap: 10px;
  grid-template-columns: repeat(1, 1fr);
}

/* Small devices (landscape phones, 576px and up) */
@media screen and (min-width: 576px) {
  .NewsContainer {
    grid-template-columns: repeat(2, 1fr);
  }
}

/* Medium devices (tablets, 768px and up) */
@media screen and (min-width: 768px) {
  .CategoryContainer {
    align-items: flex-start;
  }
  .ContentContainer {
    flex-direction: row;
  }
  .TitleContainer {
    width: 80%;
  }
  .ContentContainer {
    width: 80%;
  }
}

@media screen and (min-width: 992px) {
  .NewsContainer {
    grid-template-columns: repeat(3, 1fr);
  }
}
