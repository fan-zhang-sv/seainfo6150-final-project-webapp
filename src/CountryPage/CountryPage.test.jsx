import React from "react";
import CountryPage from "./CountryPage";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

describe("CountryPage tests", () => {
  it("renders correctly", () => {
    const { container } = render(
      <BrowserRouter>
        <CountryPage countryCode={"us"} />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
