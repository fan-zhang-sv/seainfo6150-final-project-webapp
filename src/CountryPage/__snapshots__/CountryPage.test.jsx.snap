// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryPage tests renders correctly 1`] = `
<div>
  <div
    class="Container"
  >
    <div
      class="TitleContainer"
    >
      <h1>
        United States - US
      </h1>
    </div>
    <div
      class="ContentContainer"
    >
      <div
        class="CategoryContainer"
      >
        <button
          class="CategoryButton SelectedButton"
        >
          US
        </button>
        <button
          class="CategoryButton PlainButton"
        >
          Business
        </button>
        <button
          class="CategoryButton PlainButton"
        >
          Entertainment
        </button>
        <button
          class="CategoryButton PlainButton"
        >
          Politics
        </button>
        <button
          class="CategoryButton PlainButton"
        >
          Sports
        </button>
        <button
          class="CategoryButton PlainButton"
        >
          World
        </button>
      </div>
      <div
        class="NewsContainer"
      />
    </div>
  </div>
</div>
`;
